.circleci/config.yml

```diff
@@ -113,11 +106,18 @@ jobs:
             - run:
                 name: Run UI tests against earlier supported major IntelliJ versions
                 command: ./gradlew -Pheadless -Pagainst=latestMinorsOfOldSupportedMajors uiTest
-      - store_artifacts:
-          path: build/thread-dump/
-          destination: .
+      - run:
+          name: Collect artifacts when an exception occurs
+          # language=sh
+          command: |
+            mkdir -p /exception-artifacts
+            cp /tmp/ide-probe/screenshots/uiTest/exception/* $_
+          when: on_fail
       - store_test_results:
           path: build/test-results/uiTest/
+      - store_artifacts:
+          path: /exception-artifacts
```

Comment:
Oh! it's not possible to `store_artifacts` on fail directly (without an intermediate step like the one above)?

Reply:
unfortunately ;/